datacommons_client/endpoints/payloads.py

```diff
@@ -0,0 +1,237 @@
+from abc import ABC
+from abc import abstractmethod
+from dataclasses import dataclass
+from dataclasses import field
+from enum import Enum
+from typing import Dict, Optional
+
+
+@dataclass
+class EndpointPayload(ABC):
+    """
+    Abstract base class for payload dataclasses.
+    Defines the required interface for all payload dataclasses.
+    """
+
+    @abstractmethod
+    def normalize(self) -> None:
+        """Normalize the payload for consistent internal representation."""
+        pass
+
+    @abstractmethod
+    def validate(self) -> None:
+        """Validate the payload to ensure its structure and contents are correct."""
+        pass
+
+    @property
+    @abstractmethod
+    def to_dict(self) -> dict:
+        """Convert the payload into a dictionary format for API requests."""
+        pass
+
+
+@dataclass
+class NodePayload(EndpointPayload):
+    """
+    A dataclass to structure, normalize, and validate the payload for a node API request.
```

Comment:
Can you note that this is specifically for the /v2/node endpoint?

Reply:
Same for the other payload below

Reply:
Done here: https://github.com/datacommonsorg/api-python/pull/208/commits/cda23078c42f3702295fdaff89ca445648f3cb4c